config/manifests/vllm/gpu-deployment.yaml

```diff
@@ -46,26 +46,103 @@ spec:
             - containerPort: 8000
               name: http
               protocol: TCP
+          lifecycle:
+            preStop:
+              # vLLM stops accepting connections when it receives SIGTERM, so we need to sleep
+              # to give upstream gateways a chance to take us out of rotation. The time we wait
+              # is dependent on the time it takes for all upstreams to completely remove us from
+              # rotation. Older or simpler load balancers might take upwards of 30s, but we expect
+              # our deployment to run behind a modern gateway like Envoy which is designed to 
+              # probe for readiness aggressively.
+              sleep:
```

Comment:
Worth noting this requires the `PodLifecycleSleepAction` feature gate which became Beta since 1.30

Reply:
A year ago, hrm.  Let me do some crunching to see whether that's a problem.

Reply:
I think we should set 1.30 as our minimum bar, I've included an example exec though.